### ISSUE TITLE
Dockerfile failing to build again

### ISSUE THREAD

**Issue by samcro1967** [Tzahi12345/YoutubeDL-Material#599]
Builds worked on Sunday after the updates, but failed yesterday and today.

```
Get:12 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [29.4 kB]
Get:13 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [61.3 kB]
Get:14 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [82.3 kB]
Fetched 20.6 MB in 2s (10.5 MB/s)
Reading package lists...
E: Problem executing scripts APT::Update::Post-Invoke 'rm -f /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin || true'
E: Sub-process returned an error code
The command '/bin/sh -c apt-get update && apt-get -y install   curl   gnupg   nodejs   npm &&   apt-get install -f &&   apt-get autoremove --purge &&   apt-get autoremove &&   apt-get clean &&   npm config set strict-ssl false &&   npm install -g @angular/cli' returned a non-zero code: 100
exit status 100
```

**Comment by GlassedSilver** [comment]
Well, it worked with GH's builder VM... :shrug: 

Which host are you building on? Maybe clear its caches?

**Comment by samcro1967** [comment]
Host is Ubuntu 20.04 and Docker 20.10.14.  I am running it in a pipeline with Gitea and Drone.  The Drone Runner is using the plugins/docker container to build the Dockerfile.  I might need to look for another container for the build step for this repo.

I tired to build the container directly on the host outside of the pipeline and got a different error.  Also seeing several errors (red text) in the build, but most look like cautions from what I can tell and not specific failures.  Will wait for the PR and retest as it looks liek there is a lto of movement at the moment. Thanks for the follow up.

```
Step 19/28 : WORKDIR /app
 ---> Running in b44744b8cf56
Removing intermediate container b44744b8cf56
 ---> c8cd2de1a2a4
Step 20/28 : COPY --from=ffmpeg /usr/local/bin/ffmpeg /usr/local/bin/ffmpeg
COPY failed: stat usr/local/bin/ffmpeg: file does not exist
```

**Comment by GlassedSilver** [comment]
If you get a file does not exist error for the ffmpeg part, make sure that your builder supports multi-stage docker building.

**Comment by samcro1967** [comment]
The ffmpeg error came from `docker build -t myimage .`, not the Drone Runner.  I believe `docker build` supports multi-stage already, but maybe I am missing something.

**Comment by GlassedSilver** [comment]
I know @Tzahi12345 builds images himself locally on his dev machine, I personally just commit to my own fork and have github actions build for me :D 

He might be able to clue you in on how to tweak it.

**Comment by samcro1967** [comment]
No rush on my part.  I am using a previous image that is working fine and stopped my CI/CD pipeline and nightly updates for this image so I am stable as you work toward 4.3.  Figured you would want some feedback from folks trying the updates out in the wild.

**Comment by GlassedSilver** [comment]
> No rush on my part. I am using a previous image that is working fine and stopped my CI/CD pipeline and nightly updates for this image so I am stable as you work toward 4.3. Figured you would want some feedback from folks trying the updates out in the wild.

Your help is appreciated, but atm we're constantly fixing these builds as we go anyway, so yeah.

Also, the actual "current stage" is in my repo as I said, the main repo isn't even rolling as hot as mine.
That's because that's where the PR will be coming from that hopefully fixes everything.

**Comment by GlassedSilver** [comment]
I'll link the PR which should fix it. Once the PR merges this will be closed and marked as resolved.

Should you continue to see this issue on the THEN most recent build in the nightly-branch please do not hesitate to comment here, and we will re-open and re-investigate.

Hope you have a wonderful week so far and let's hope this is finally checked off. :)

**Comment by GlassedSilver** [comment]
Just some additional information for anyone who may still want to use Drone's Docker runner, [from their docs](https://docs.drone.io/runner/docker/overview/):

> **When to Avoid?**
> 
> The Docker runner is poorly suited for projects that cannot run tests or compile code inside containers, including projects that target operating systems or architectures not supported by Docker, such as macOS.
> 
> The Docker runner is also poorly suited for stateful pipelines that need to store files or folders on the host machine in-between pipeline executions. Docker pipelines are ephemeral and do not mutate the host machine.

**Comment by samcro1967** [comment]
It looks like there are a couple of ways to do multistage with Drone.  Agreed though it should be supported more natively and out of the box without jumping through extra steps.

I ran the build this morning (forgot to disable daily rebuilds) and it completed successfully, but I did not try to run it to validate the image created works successfully.  So the changes to Dockerfile over the last couple of days seem to have at least the build working again in my environment.

[How to effectively use cached layers for docker build steps?](https://community.harness.io/t/how-to-effectively-use-cached-layers-for-docker-build-steps/10435)

**Option 1 - caching layer**
[The ultimate DroneCI caching guide](https://laszlo.cloud/the-ultimate-droneci-caching-guide#how-to-use-docker-layer-caching-as-the-primary-caching)

**Options 2 - Host Docker daemon**
[Interact directly with the host machine docker daemon](https://docs.drone.io/pipeline/docker/examples/services/docker/)
